src/mesh_analysis/readers/file_reader.py

```diff
@@ -0,0 +1,56 @@
+# Python Imports
+import logging
+import multiprocessing
+import pandas as pd
+from typing import List
+from pathlib import Path
+
+# Project Imports
+from src.utils import file_utils
+from src.mesh_analysis.readers.reader import Reader
+from src.mesh_analysis.tracers.message_tracer import MessageTracer
+
+
+logger = logging.getLogger(__name__)
+
+
+class FileReader(Reader):
+
+    def __init__(self, folder: str, tracer: MessageTracer):
+        self._folder = folder
+        self._tracer = tracer
+
+    def read(self) -> pd.DataFrame:
+        logger.info(f'Reading {self._folder}')
+        files_result = file_utils.get_files_from_folder_path(Path(self._folder))
+
+        if files_result.is_ok():
+            parsed_logs = self._read_files(files_result.ok_value)
+            logger.info(f'Tracing {self._folder}')
+            df = self._tracer.trace(parsed_logs)
+        else:
+            logger.error(f'Could not read {self._folder}')
+            exit()
+
+        return df
```

Comment:
Styling opinion: When one of the `if` arms makes an early exit, I generally favour doing that check first; but I'm fine as it is :)
```suggestion
        if not files_result.is_ok():
            logger.error(f'Could not read {self._folder}')
            exit()

        parsed_logs = self._read_files(files_result.ok_value)
        logger.info(f'Tracing {self._folder}')
        return self._tracer.trace(parsed_logs)
```